activesupport/lib/active_support/cache.rb

```diff
@@ -718,6 +719,17 @@ def merged_options(call_options)
             expires_at = call_options.delete(:expires_at)
             call_options[:expires_in] = (expires_at - Time.now) if expires_at
 
+            if call_options[:expires_in]&.negative?
+              error = ArgumentError.new("Cache expiration time is invalid")
```

Comment:
Best practice is to include the invalid value in the error message. It often makes it much easier to debug.

Reply:
Also you want to clearly state it's invalid because it's negative